src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -162,6 +162,18 @@ public void RegisterOperationAction(Action<OperationAnalysisContext> action, par
         /// <param name="action">Action to be executed at completion of semantic analysis of an <see cref="IOperation"/>.</param>
         /// <param name="operationKinds">Action will be executed only if an <see cref="IOperation"/>'s Kind matches one of the operation kind values.</param>
         public abstract void RegisterOperationAction(Action<OperationAnalysisContext> action, ImmutableArray<OperationKind> operationKinds);
+
+        /// <summary>
+        /// Register for concurrent execution of analyzer actions registered by this analyzer.
+        /// An analyzer that registers for concurrent execution should be more performant then a non-concurrent analyzer.
+        /// However, such analyzers must ensure that its actions are implemented in a thread-safe manner.
```

Comment:
Maybe "Such an analyzer must ensure that its actions can execute correctly in parallel"? I've heard enough arguments about the precise meaning of "thread safe" that I avoid the term.

At least, "analyzers" and "its" must be reconciled.